static/js/message_list_data.js

```diff
@@ -178,6 +178,18 @@ export class MessageListData {
     }
 
     messages_filtered_for_user_mutes(messages) {
+        if (this.filter === undefined || this.filter.is_non_huddle_pm === undefined) {
+            // Actually these will never be undefined, but this condition makes it possible for
+            // our test suite modules (which are testing things other than muting) to create message
+            // lists using dummy or undefined filter objects.
+            return [...messages];
```

Comment:
This part is a bit hacky. The need to do this is other tests involving message lists fail with this change if they haven't defined a `filter` object for the list.
I thought of stubing `message_list_data.messages_filtered_for_user_mutes` instead -- would that be better?

Reply:
How many tests use this trick?  If it's not hard, the best option would probably be for message_list objects in tests to always have a filter object, if it's only a bit of stubbing work?  Second best is the stubbing approach.

Reply:
Turns out not many. It was actually a nice opportunity to remove some unnecessary mocks 🙂. I added a prep commit to do so.